### ISSUE TITLE
Smart Lock does not retrieve saved Facebook credential

### ISSUE THREAD

**Issue by wuman** [firebase/FirebaseUI-Android#171]
The current implementation (as of commit c9962e7) uses the `CredentialsAPI` class to request for saved credentials in Smart Lock.  However, it does not appear to retrieve any other account types other than Google.  The end result is that a saved Facebook credential cannot be retrieved for further auto sign-in.

This can seemingly be fixed by requiring the `CredentialsAPI` class depend on a list of account types and then requesting for the credentials of those account types in `CredentialsAPI#requestCredentials()`:

```
public void requestCredentials(final boolean shouldResolve, boolean onlyPasswords) {
    CredentialRequest.Builder crBuilder = new CredentialRequest.Builder().setPasswordLoginSupported(true);

    if (!onlyPasswords) {
        crBuilder.setAccountTypes(IdentityProviders.GOOGLE, IdentityProviders.FACEBOOK);
    }

    ...
}
```

I'd love to send a pull request for this patch, but would like to first raise it here to see if there is a reasoning behind retrieving only the Google account type that I have overlooked in the grand scheme of things.

**Comment by samtstern** [comment]
@wuman a pull request is always welcome, thanks for the detailed issue report.
@amandle do you know why the code is currently this way?

**Comment by SUPERCILEX** [comment]
Fixed in #357.

**Comment by amandle** [comment]
@samtstern I don't know why it was written that wa. It's strange since we have had Facebook support since the beginning.  I will send a PR to fix this for this version of the code, and @SUPERCILEX's fix will land when #357 lands
